lib/core/src/lib/form/components/widgets/date-time/date-time.widget.ts

```diff
@@ -42,68 +43,115 @@ import { WidgetComponent } from '../widget.component';
     ],
     templateUrl: './date-time.widget.html',
     styleUrls: ['./date-time.widget.scss'],
-    imports: [NgIf, TranslateModule, MatFormFieldModule, MatInputModule, MatDatetimepickerModule, FormsModule, ErrorWidgetComponent],
+    imports: [NgIf, TranslateModule, MatFormFieldModule, MatInputModule, MatDatetimepickerModule, ReactiveFormsModule, ErrorWidgetComponent],
     encapsulation: ViewEncapsulation.None
 })
-export class DateTimeWidgetComponent extends WidgetComponent implements OnInit {
+export class DateTimeWidgetComponent extends WidgetComponent implements OnInit, OnDestroy {
     minDate: Date;
     maxDate: Date;
+    datetimeInputControl: FormControl<Date>;
```

Comment:
either provide default values for the properties or declare them with `?`

Reply:
we need to define this control at the ngOnInit level, because then we have access to the initial value which is the default value of the control. so in that case, shouldn't we use exclamation mark `!` instead making it opitonal by `?`?

Reply:
You can define the control here and patch it in the ngOnInit when the value is defined. 
Also actually it's another problem, because afaik, value is an input. So to make it 100% we should patch the form every time the value is changed.... given that the base implementation is correct 😅

Reply:
defined at declaration level, and added patch in the onInit.
about onChange, i think its not needed (i checked it), because we don't update `this.field` during component lifecycle, so ngOnChanges is not called for this input. Its only used to initialize default values